src/QueryM/DatumCacheWsp.purs

```diff
@@ -104,24 +106,56 @@ instance DecodeAeson GetDatumsByHashesR where
   decodeAeson r =
     let
       decodeDatumArray
-        :: Aeson -> Either JsonDecodeError (Map DataHash Datum)
+        :: Aeson -> Either JsonDecodeError (Map DataHash (Either String Datum))
       decodeDatumArray =
         caseAesonArray (Left $ TypeMismatch "expected array")
           $ (map Map.fromFoldable) <<< traverse decodeDatum
 
+      decodeEmptyDatums obj =
+        pure Map.empty *>
+          ( caseAesonArray (Left $ TypeMismatch "expected array")
+              $ identity <<< traverse decodeEmptyDatum
+          ) obj
+
+      decodeEmptyDatum
+        :: Aeson -> Either JsonDecodeError Unit
+      decodeEmptyDatum aes =
+        if stringifyAeson aes == "{}" then
+          pure unit
+        else
+          Left $ TypeMismatch "expected empty JSON"
```

Comment:
```suggestion
      decodeEmptyDatum aes =
        unless (caseAesonObject false isEmpty aes) (Left $ TypeMismatch "Empty Object")
```

Reply:
yes, I'm looking for an `isEmpty` function like this, but It doesn't seem to exist in the current Aeson api.

Reply:
It's in Foreign.Object, which I believe is what the Object type here is (can't check ATM)

Reply:
https://pursuit.purescript.org/packages/purescript-foreign-object/4.0.0/docs/Foreign.Object#v:isEmpty

Reply:
Thanks! I didn't know I can use it like this. Sadly this isn't needed anymore with the last change in ODC api.